skada/deep/base.py

```diff
@@ -680,7 +692,9 @@ def score(
             The mean accuracy score.
         """
         X, _ = self._prepare_input(X, sample_domain, sample_weight)
-        return super().score(X, y, **score_params)
+        X['X'], X['sample_domain'] = check_X_domain(X['X'], sample_domain=X['sample_domain'], allow_nd=True, allow_source=allow_source)
+
+        return accuracy_score(y, self.predict(X, sample_domain, allow_source=allow_source), sample_weight=sample_weight)
```

Comment:
Because the implementation of `super().score` is:
`return accuracy_score(y, self.predict(X), sample_weight=sample_weight)`
Thus we can't feed the `allow_source` to the `self.predict`